src/test/java/dev/snowdrop/release/services/CPaaSConfigUpdateServiceTest.java

```diff
@@ -0,0 +1,190 @@
+ * specific language governing permissions and limitations under the License.
+ */
+package dev.snowdrop.release.services;
+
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import dev.snowdrop.release.model.POM;
+import dev.snowdrop.release.model.Release;
+import dev.snowdrop.release.model.cpaas.product.CPaaSProduct;
+import dev.snowdrop.release.model.cpaas.product.CPaaSProductFile;
+import dev.snowdrop.release.model.cpaas.release.CPaaSAdvisory;
+import dev.snowdrop.release.model.cpaas.release.CPaaSPipelines;
+import dev.snowdrop.release.model.cpaas.release.CPaaSRelease;
+import dev.snowdrop.release.model.cpaas.release.CPaaSReleaseFile;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author <a href="antcosta@redhat.com">Antonio Costa</a>
+ */
+@QuarkusTest
+@TestProfile(TestProfiles.IntegrationTests.class)
+public class CPaaSConfigUpdateServiceTest {
+
+    private static final String RELEASE = "2.4.0";
+    private static final String PREVIOUS_RELEASE = "2.3.6";
+    private static final String PRODUCT_FILE_NAME = "product.yml";
+    private static final String RELEASE_FILE_NAME = "release.yml";
+    private static final YAMLMapper MAPPER = new YAMLMapper();
+
+    @Inject
+    CPaaSConfigUpdateService service;
+
+    @Inject
+    CPaaSReleaseFactory factory;
+
+    @Inject
+    GitService git;
+
+    @Test
+    public void testProduct() {
+        final String token = ConfigProvider.getConfig().getValue("gitlab.token", String.class);
+        final String user = ConfigProvider.getConfig().getValue("gitlab.user", String.class);
+        GitService.GitConfig config = service.buildGitConfig(RELEASE, user, token, Optional.of(PREVIOUS_RELEASE), Optional.of(ConfigProvider.getConfig().getOptionalValue("gitlab.repository", String.class).orElse(user + "/springboot")));
+        try {
+            git.initRepository(config);
+            Stream<File> files = service.updateCPaaSFiles(RELEASE, git.getConfig(config).getRepository().getWorkTree(), PREVIOUS_RELEASE, false, true);
+            Map<String, File> fileMap = files.collect(Collectors.toMap(file -> file.getName(), file -> file));
+            assertTrue(fileMap.containsKey(PRODUCT_FILE_NAME), fileMap.toString());
+            testProductFile(factory.createCPaaSProductFromStream(new FileInputStream(fileMap.get(PRODUCT_FILE_NAME))));
+        } catch (IOException | ExecutionException | InterruptedException e) {
```

Comment:
I prefer to add exception to the method signature (`throws Exception`), the test will fail also and the code is more readable.

Reply:
Done.